src/librustdoc/html/static/main.js

```diff
@@ -360,15 +368,68 @@
                 }
             }
 
+            function extractGenerics(val) {
+                var values = val.split("<")
+                                .slice(1)
+                                .join("")
+                                .split(">")
+                                .slice(0, -1)
+                                .join("")
+                                .split(",");
+                for (var x = 0; x < values.length; ++x) {
+                    values[x] = values[x].trim();
+                }
+                return values;
+            }
+
             function findArg(obj, val) {
+                var isWrapper = val.indexOf('<') !== -1;
+                var lev_distance = MAX_LEV_DISTANCE + 1;
                 if (obj && obj.type && obj.type.inputs.length > 0) {
                     for (var i = 0; i < obj.type.inputs.length; i++) {
                         if (obj.type.inputs[i].name === val) {
-                            return true;
+                            // No need to check anything else: we found it. Let's just move on.
+                            return 0;
+                        }
+                        var name = obj.type.inputs[i].name.toLowerCase()
+                        var tmp_val = [name];
+                        if (name.indexOf('<') === -1 && isWrapper === true) {
+                            tmp_val = extractGenerics(name);
```

Comment:
Do you mean to use `name` here? The check above says `name` won't contain any `'<'`...

(If this is expected why not just write `tmp_valu = name.split(/\s*,\s*/)`...)

Reply:
Or maybe not... Trying to understand what I was trying to do...

Reply:
No I was right: it should have been `!== -1`!

Reply:
Wait a sec, this is ripping out the contents of the generics *if both the type and the search query contain them*? Am i reading this correctly? It seems like we should do that if `isWrapper === false` instead.